src/pages/Search/SearchAdvancedFiltersPage/SearchFiltersExportedPage/index.tsx

```diff
@@ -0,0 +1,104 @@
+import React, {useCallback, useMemo, useState} from 'react';
+import {useOnyx} from 'react-native-onyx';
+import {updateAdvancedFilters} from '@libs/actions/Search';
+import Navigation from '@libs/Navigation/Navigation';
+import type {SearchDateModifier} from '@libs/SearchUIUtils';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import SearchFiltersExportedCalendarView from './CalendarView';
+import SearchFiltersExportedRootView from './RootView';
+
+type SearchFiltersExportedPageValues = Record<SearchDateModifier, string | null>;
+
+function SearchFiltersExportedPage() {
+    const [searchAdvancedFiltersForm] = useOnyx(ONYXKEYS.FORMS.SEARCH_ADVANCED_FILTERS_FORM, {canBeMissing: true});
+
+    // Get all of the initial values from the advanced filters form and store them locally so we can modify
+    // them without mutating the original form
+    const initialState = useMemo(() => {
+        return Object.values(CONST.SEARCH.DATE_MODIFIERS).reduce((acc, dateType) => {
+            acc[dateType] = searchAdvancedFiltersForm?.[`${CONST.SEARCH.SYNTAX_FILTER_KEYS.EXPORTED}${dateType}`] ?? null;
+            return acc;
+        }, {} as SearchFiltersExportedPageValues);
+        // The initial value of a useState only gets calculated once, so we dont need to keep calculating this initial state
+        // eslint-disable-next-line react-compiler/react-compiler
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    const [view, setView] = useState<SearchDateModifier | null>(null);
+    const [localDateValues, setLocalDateValues] = useState<SearchFiltersExportedPageValues>(initialState);
```

Comment:
```suggestion
    const [view, setView] = useState<SearchDateModifier | null>(null);
    const [localDateValues, setLocalDateValues] = useState<SearchFiltersExportedPageValues>(() => {
        // Get all of the initial values from the advanced filters form and store them locally so we can modify
        // them without mutating the original form
        return Object.values(CONST.SEARCH.DATE_MODIFIERS).reduce((acc, dateType) => {
            acc[dateType] = searchAdvancedFiltersForm?.[`${CONST.SEARCH.SYNTAX_FILTER_KEYS.EXPORTED}${dateType}`] ?? null;
            return acc;
        }, {} as SearchFiltersExportedPageValues);
        // The initial value of a useState only gets calculated once, so we dont need to keep calculating this initial state
        // eslint-disable-next-line react-compiler/react-compiler
        // eslint-disable-next-line react-hooks/exhaustive-deps
    });
```

`useState` allows you to pass a callback as the initial state, the callback will only be executed the first time to initialize it, there is no need for this `useMemo` workaround to achieve the same

Reply:
I didn't know that, very helpful, thanks!